### PR TITLE
test ubuntu as well as centos nodes in integration tests

### PR DESCRIPTION
**Pull request by D3nn**
tests both ubuntu and centos as nodes in cluster.

### CODE REVIEW

File: test/integration/container/multimaster_test.go [hunk 1]
```diff
@@ -284,86 +378,101 @@ func TestMultimasterSetup(t *testing.T) {
...
+	}
 
-	for i := 0; i < 4; i++ {
-		for _, kubeletArg := range expectedKubeletArgs {
-			log.Infof("Checking kubelet arg (%s) on node%d", kubeletArg, i)
-			run(t, "footloose",
-				"-c", "../../../examples/footloose/"+node_os+node_version+"/docker/multimaster.yaml",
-				"ssh", fmt.Sprintf("root@node%d", i), fmt.Sprintf("ps -ef | grep -v 'ps -ef' | grep /usr/bin/kubelet | grep %s", kubeletArg))
+	for _, tc := range tests {
+		tc := tc
```
Comment (jrryjcksn) [inline test/integration/container/multimaster_test.go:400]: This seems like some leftover cruft...

File: test/integration/container/multimaster_test.go [hunk 2]
```diff
@@ -284,86 +378,101 @@ func TestMultimasterSetup(t *testing.T) {
...
+	}{
+		{
+			name: CENTOS,
+			path: "../../../examples/footloose/centos7/docker/multimaster.yaml",
+			skip: false,
+		},
+		{
+			name: UBUNTU,
+			path: "../../../examples/footloose/ubuntu1804/docker/multimaster.yaml",
+			skip: true,
```
Comment (jrryjcksn) [inline test/integration/container/multimaster_test.go:395]: How do we select Ubuntu (or get both to run) now?

File: test/integration/container/multimaster_test.go (hunk 1 above)
Comment (D3nn) [reply test/integration/container/multimaster_test.go:402]: This is the section that actually runs the tests...

File: test/integration/container/multimaster_test.go (hunk 2 above)
Comment (D3nn) [reply test/integration/container/multimaster_test.go]: It should still be possible to select the node_os you'd like to run.  There are two ways to go about it:

1. `NODE_OS=<ubuntu|centos> go test -v test/integration/container...`
2. `NODE_OS=<ubuntu|centos> make integration-tests-container`

**Comment by bboreham** [comment]
We already test Ubuntu in integration tests, e.g. [for this PR](https://app.circleci.com/pipelines/github/weaveworks/wksctl/1228/workflows/fc588606-3f71-4a13-b472-54c70283a215/jobs/7852)

What is the justification to add them to this particular test?

**Comment by bboreham** [comment]
Does this PR replace #220 ?

File: Makefile [hunk 3]
```diff
@@ -135,10 +135,16 @@ container-tests: pkg/apis/wksprovider/machine/scripts/scripts_vfsdata.go pkg/api
...
+# Valid values are centos or ubuntu, e.g NODE_OS=ubuntu make integration-tests-container
+
+# Leaving NODE_OS unset will result in one of centos or ubuntu being chosen for the test
+
 	NODE_OS1="centos"
 	NODE_OS2="ubuntu"
-	NODE_OS_CHOICE=$${NODE_OS_CHOICE:-"$$(echo $$NODE_OS1 $$NODE_OS2 | tr ' ' '\n' | shuf | head -1)"}
-	IMAGE_TAG=$(IMAGE_TAG) NODE_OS="$${NODE_OS_CHOICE}" go test -v -timeout 40m ./test/integration/container/...
+	NODE_OS_CHOICE=$${NODE_OS:-"$$(echo $$NODE_OS1 $$NODE_OS2 | tr ' ' '\n' | shuf | head -1)"}
+	IMAGE_TAG=$(IMAGE_TAG) NODE_OS="$${NODE_OS}" go test -v -timeout 40m ./test/integration/container/...
```
Comment (jrryjcksn) [inline Makefile:147]: I believe the second use of NODE_OS here should be NODE_OS_CHOICE

**Review by bboreham — CHANGES_REQUESTED** [review]
Blocking merge until someone explains why we need this.

File: Makefile (hunk 3 above)
Comment (D3nn) [reply Makefile]: Have replaced NODE_OS_CHOICE with just NODE_OS

**Comment by D3nn** [comment]
> Does this PR replace #220 ?

It does.

**Comment by D3nn** [comment]
> Blocking merge until someone explains why we need this.

The biggest reason is that we can run such tests on our local machines as it is much more efficient when we don't have to push to GH, check test output, and address issues and possibly iterate some more to get the tests and/or code working correctly.

**Review by bboreham — APPROVED** [review]
> much more efficient

A smaller test would meet that aim better - firing up 3x control-plane nodes and one worker is quite a lot more work than we need just to see if our Ubuntu install is working.

Also we should have a local apt repository, as we do for yum.

"approve" to undo my earlier block; I haven't actually read the code.

**Comment by D3nn** [comment]
Since the test is a "multimaster" test (for both centos and ubuntu), it
seems like we'd want a 3:1 cluster.  No?

We might be able to do a 2:1 cluster, but somehow that just feels wrong.  🤔

**Comment by bboreham** [comment]
"I want an Ubuntu test" is satisfied by a 1-node cluster.
"I want a multimaster test" is satisfied by a Centos test.

I haven't heard any argument why these have to be combined.

**Comment by bboreham** [comment]
Please make unrelated changes in separate PRs; it makes it easier to figure out what happened.
(Specifically the rename of `config/crd/cluster.weave.works_existinginframachines.yaml`.)

**Pull request merged**